src/coreclr/jit/target.h

```diff
@@ -208,8 +208,163 @@ enum _regMask_enum : unsigned
 // In any case, we believe that is OK to freely cast between these types; no information will
 // be lost.
 
-#if defined(TARGET_AMD64) || defined(TARGET_ARMARCH) || defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
+typedef _regNumber_enum regNumber;
+typedef unsigned char   regNumberSmall;
+
+#if defined(TARGET_AMD64) || defined(TARGET_ARM) || defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
 typedef unsigned __int64 regMaskTP;
+#elif defined(TARGET_ARM64)
+typedef unsigned __int64 regMaskSmall;
+struct _regMaskAll
+{
+    unsigned __int64 low;
+
+    _regMaskAll(unsigned __int64 regMask)
+        : low(regMask)
+    {
+    }
+
+    _regMaskAll()
+        : low(RBM_NONE)
+    {
+    }
+
+    FORCEINLINE operator _regMaskAll() const
+    {
+        return _regMaskAll{static_cast<uint64_t>(low)};
+    }
```

Comment:
What is this for?

Reply:
copied it over from my previous prototype. Turns out this is not needed.